flink-core/src/main/java/org/apache/flink/api/connector/sink/SinkWriter.java

```diff
@@ -63,6 +63,9 @@ default void writeWatermark(Watermark watermark) throws IOException, Interrupted
      *
      * <p>This will be called before we checkpoint the Writer's state in Streaming execution mode.
      *
+     * <p>In case the sink implements a 1-phase commit protocol, this method should trigger the
```

Comment:
I wonder if this is really self-explanatory. How about

```suggestion
<p>In case the sink has no explicit committer, this method is still called to allow the writer to implement a 1-phase commit protocol. 
```

Reply:
Sorry for being nitty, but I'd like to have the interfaces in good shape.

Reply:
No worries, your suggestion definitely provides more information about the committer writer relation.